lib/appsignal/logger.ex

```diff
@@ -1,120 +1,52 @@
 defmodule Appsignal.Logger do
   require Appsignal.Utils
 
-  @io Appsignal.Utils.compile_env(:appsignal, :io, IO)
-  @file_module Appsignal.Utils.compile_env(:appsignal, :file, File)
-
-  @log_levels [:trace, :debug, :info, :warn, :error]
+  @nif Appsignal.Utils.compile_env(:appsignal, :appsignal_tracer_nif, Appsignal.Nif)
+  @severity %{
+    trace: 1,
+    debug: 2,
+    info: 3,
+    warn: 5,
+    error: 6
+  }
 
   @type log_level :: :trace | :debug | :info | :warn | :error
-  @type device :: :stdio | :stderr | :file
-
-  @spec trace(String.t()) :: :ok
-  def trace(message) do
-    log(:trace, message, [])
-  end
-
-  @spec debug(String.t()) :: :ok
-  def debug(message) do
-    log(:debug, message, [])
-  end
-
-  @spec info(String.t()) :: :ok
-  def info(message) do
-    log(:info, message, [])
-  end
-
-  @spec warn(String.t()) :: :ok
-  def warn(message, options \\ []) do
-    log(:warn, message, options)
-  end
-
-  @spec error(String.t()) :: :ok
-  def error(message, options \\ []) do
-    log(:error, message, options)
-  end
 
-  @spec log(log_level(), String.t(), keyword()) :: :ok
-  defp log(level, message, options) do
-    threshold = Appsignal.Config.log_level()
-
-    if log_level?(level, threshold) do
-      case {device(), Keyword.get(options, :stderr, false)} do
-        {device, false} ->
-          do_log(device, level, message)
-
-        {:stdio, true} ->
-          do_log(:stderr, level, message)
-
-        {:file, true} ->
-          do_log(:file, level, message)
-          do_log(:stderr, level, message)
-      end
-    end
+  @spec trace(String.t(), String.t(), %{}) :: :ok
+  def trace(group, message, metadata \\ %{}) do
+    log(:trace, group, message, metadata)
```

Comment:
For Ruby, we support the log levels that the standard Ruby logger support. (For Node.js, there's no standard logger, so we use the Rust ones instead)

Would it be possible to use for Elixir [the standard log levels that the Elixir logger supports](https://hexdocs.pm/logger/1.14/Logger.html#module-levels)? This would mean removing `trace` and adding a bunch of missing ones. The levels should already [be supported by the agent logging PR](https://github.com/appsignal/appsignal-agent/pull/847/files#diff-31519f6f697f6242b33b48dbf34dc2309d6048713669a0bcf8a8249b9cdb7b85R76-R83), except that `emergency` is called `fatal`.

Reply:
I assumed we were going for the same levels for the three integrations as they will map to what is seen in AppSignal. I'll remap them to the language standard then.

Reply:
Done! Thank you.

Reply:
No prob! Also, I'd be happy to reconsider the log levels for Node.js -- the whole "there's no standard logger so I'll use Rust's levels" is just something I came up with.

Maybe we want to consider `console` the standard logger?

Reply:
These are the npm "official" log levels, maybe we can use them.

```js
{
  error: 0,
  warn: 1,
  info: 2,
  http: 3,
  verbose: 4,
  debug: 5,
  silly: 6
}
```

Ref: https://docs.npmjs.com/cli/v8/using-npm/logging